### PR TITLE
#27 put url into fields of raw_attribute for ddx ingester

### LINKED ISSUE #14: Set a deterministic entry_id when harvesting
The value of the `entry_id` field should be unique and characterize the dataset rather than being generated randomly.
The idea is to be able to find out if the same dataset comes from different URLs.

This needs to be done for each ingester.

### LINKED ISSUE #27: remove the temporary solution in ftp harvester!
There is a temporary solution in ftp harvester for setting the entry_id of each dataset. I must be removed and the correct place for the solution is inside the metanorm repo. This related to the issue https://github.com/nansencenter/metanorm/issues/22#issue-695801954

### PR DESCRIPTION
**Pull request by opsdep**
closes #27 
closes #14 
@akorosov 
@aperrin66 
I put url into fields of raw_attribute for ddx ingester. for the copernicus, the things are done in the metanorm. 

Here once again we have first merge the pr of metanorm and then merge this pr in order to working properly.

Everything is ready with these files and I won't change them unless you tell me your review.

### CODE REVIEW

File: geospaas_harvesting/ingesters.py [hunk 1]
```diff
@@ -373,14 +374,14 @@ def prepare_url(url):
 
     def _get_normalized_attributes(self, url, *args, **kwargs):
         """Get normalized metadata from the DDX info of the dataset located at the provided URL"""
-
+        original_url = url
```
Comment (akorosov) [inline geospaas_harvesting/ingesters.py]: You don't modify the variable `url` below. Don`t create a new one.
```suggestion
```

File: geospaas_harvesting/ingesters.py [hunk 2]
```diff
@@ -373,14 +374,14 @@ def prepare_url(url):
...
         prepared_url = self.prepare_url(url)
         # Get the metadata from the dataset as an XML tree
         stream = io.BytesIO(requests.get(prepared_url, stream=True).content)
         # Get all the global attributes of the Dataset into a dictionary
         extracted_attributes = self._extract_attributes(
             ET.parse(stream).getroot())
-
+        self.add_url(original_url, extracted_attributes)
```
Comment (akorosov) [inline geospaas_harvesting/ingesters.py]:
```suggestion
        self.add_url(url, extracted_attributes)
```

File: geospaas_harvesting/ingesters.py [hunk 3]
```diff
@@ -446,7 +447,7 @@ def _get_normalized_attributes(self, url, *args, **kwargs):
         normalized_attributes['geospaas_service_name'] = 'ftp'
         normalized_attributes['geospaas_service'] = 'ftp'
         #Temporary solution
-        normalized_attributes['entry_id'] = urlparse(url).path.split('/')[-1]#'entry_id'is file name
+        #normalized_attributes['entry_id'] = urlparse(url).path.split('/')[-1]#'entry_id'is file name
```
Comment (akorosov) [inline geospaas_harvesting/ingesters.py]: Don't leave commented lines in the code. Always remember to delete them.
```suggestion
```

**Review by akorosov — CHANGES_REQUESTED** [review]

**Comment by opsdep** [comment]
@aperrin66 
@akorosov 
Everything is addressed in cefaa8e.

@aperrin66 
@akorosov 
You can review these code which is completely separated from the metanorm code. It is ready to be reviewed.

File: tests/test_ingesters.py [hunk 4]
```diff
@@ -290,15 +290,13 @@ def test_ingest_from_metadata_string_geometry(self):
...
 class DDXIngesterTestCase(django.test.TestCase):
     """Test the DDXIngester"""
 
     TEST_DATA = {
         'full_ddx': {
-            'url': "https://test-opendap.com/full_dataset.nc.ddx",
-            'file_path': "data/opendap/full_ddx.xml"},
-        'full_ddx_2': {
-            'url': "https://test-opendap2.com/full_dataset.nc.ddx",
+            'url': "https://opendap.jpl.nasa.gov/opendap/Afull_dataset.nc.ddx",
```
Comment (aperrin66) [inline tests/test_ingesters.py]: why the `A`?

Comment (opsdep) [reply tests/test_ingesters.py]: fixed in 61740e4. It was just a name and does not do anything in coding! So there is no particular reason for doing it   :)

File: geospaas_harvesting/ingesters.py [hunk 5]
```diff
@@ -446,7 +446,6 @@ def _get_normalized_attributes(self, url, *args, **kwargs):
         normalized_attributes['geospaas_service_name'] = 'ftp'
         normalized_attributes['geospaas_service'] = 'ftp'
         #Temporary solution
-        normalized_attributes['entry_id'] = urlparse(url).path.split('/')[-1]#'entry_id'is file name
```
Comment (aperrin66) [inline geospaas_harvesting/ingesters.py:449]: please also remove the comment

Comment (opsdep) [reply geospaas_harvesting/ingesters.py:449]: Done in 50b577a. Thanks! I have forgotten that one!

**Review by aperrin66 — DISMISSED** [review]

**Review by akorosov — DISMISSED** [review]

**Review by aperrin66 — DISMISSED** [review]

**Review by aperrin66 — APPROVED** [review]

**Pull request merged**